is-car-dag-complete.js

```diff
@@ -107,17 +113,29 @@ export class S3CarReader {
       MaxKeys: 1
     })
     const response = await this._s3.send(command)
-    console.log(response.Contents)
     return response.Contents
   }
 
   /**
+   * read a car file from s3 path and write its blocks to the provided blockstore
    * @private
    * @param {string} path
-   * @param {Blockstore blockstore
+   * @param {Blockstore} blockstore
    */
   async _readCar (path, blockstore) {
-
+    const getCarCommand = new GetObjectCommand({
+      Bucket: this._bucketName,
+      Key: path,
+    })
+    const response = await this._s3.send(getCarCommand)
+    const responseCarReader = await CarReader.fromIterable(response.Body);
+    const roots = await responseCarReader.getRoots()
+    debugLog('_readCar read car', { roots, blocks: responseCarReader.blocks() });
+    for await (const block of responseCarReader.blocks()) {
+      debugLog('_readCar writing to blockstore', block.cid.toString())
+      blockstore.set(block.cid.toString(), block);
```

Comment:
It's possible that this needs to read something to find yet more cids that should go in the blockstore than just the ones it gets from `responseCarReader.blocks()` ? (e.g. is `blocks()` walking a graph for me or do I need to do that manually?)